src/test/java/com/google/sps/servlets/LinkServletTest.java

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.HttpURLConnection;
+import java.util.Collections;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+
+@RunWith(JUnit4.class)
+public final class LinkServletTest {
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  private LinkShortenServlet linkShortenServlet;
+  private LinkServlet linkServlet;
+  private String reportsJson;
+  private StringWriter stringWriter;
+  private PrintWriter writer;
+
+  // Maximum eventual consistency.
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig()
+          .setDefaultHighRepJobPolicyUnappliedJobPercentage(100));
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    linkShortenServlet = new LinkShortenServlet();
+    linkServlet = new LinkServlet();
+    helper.setUp();
+    reset();
+    linkServlet.doGet(request, response);
```

Comment:
Setup is used to set up the pre-testing states/configs, such as variables initializations.
It's clearer to move this doGet to each test case, since it is what we do want to test against and readers would find it more readable to see it's explicitly called in each case.